### PR TITLE
7112 payara5 aio

### LINKED ISSUE #7112: switch docker-aio to payara5
Analogous to https://github.com/IQSS/dataverse/issues/6849

### PR DESCRIPTION
**Pull request by pameyer**
**What this PR does / why we need it**:

switch docker-aio from glassfish to payara

**Which issue(s) this PR closes**:

Closes #7112 

**Special notes for your reviewer**:

Deferred testing of / changes related to new install script (despite describing it as analogous to an issues that explicitly included testing the new installer); focused on API test suite.

**Suggestions on how to test this**:

Existing instructions for docker-aio should work unchanged; Server header from `curl -i http://localhost:8083/api/info/version` should report payara

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

nope

**Is there a release notes update needed for this change?**:

nope

**Additional documentation**:

_should_ be none necessary

### CODE REVIEW

File: conf/docker-aio/0prep_deps.sh [hunk 1]
```diff
@@ -3,18 +3,11 @@ if [ ! -d dv/deps ]; then
...
-	unzip glassfish-4.1.zip
-	rm glassfish4/glassfish/modules/weld-osgi-bundle.jar
-	mv weld-osgi-bundle-2.2.10.Final-glassfish4.jar glassfish4/glassfish/modules
-	tar zcf $wdir/dv/deps/glassfish4dv.tgz glassfish4
-	cd $wdir
-	# assuming that folks usually have /tmp auto-clean as needed
+
+if [ ! -e dv/deps/payara-5.2020.2.zip ]; then
+	echo "payara dependency prep"
+	# no more fiddly patching :)
```
Comment (pdurbin) [inline conf/docker-aio/0prep_deps.sh:9]: Hooray! 🎉

File: conf/docker-aio/c7.dockerfile [hunk 2]
```diff
@@ -52,8 +51,8 @@ EXPOSE 9009
 
 RUN mkdir /opt/dv
 
-# yeah - still not happy if glassfish isn't in /usr/local :<
-RUN ln -s /opt/glassfish4 /usr/local/glassfish4
+# keeping the symlink on the off chance that something else is still assuming /usr/local/glassfish4
+RUN ln -s /opt/payara5 /usr/local/glassfish4
```
Comment (pdurbin) [inline conf/docker-aio/c7.dockerfile:55]: I assume it's safe to not create this symlink. But it doesn't hurt to leave it in. I'm glad the comment was updated to described what's going on.

**Comment by pameyer** [comment]
Found that this does not work on a fresh system due to absence of glassfish-setup.sh ; was accidentally working on a dev system that still had a cached copy.

**Comment by pdurbin** [comment]
> absence of glassfish-setup.sh

It's still there but it was renamed to as-setup.sh.

**Comment by pameyer** [comment]
updated and re-tested

**Comment by pdurbin** [comment]
I'm on 6a23496 and I'm getting `./install: line 22: ./glassfish-setup.sh: No such file or directory`

@pameyer I'm wondering if you have an unpushed commit?

(Also, now you're making me refresh myself on how to stop and delete Docker containers. 😄 )

**Comment by pameyer** [comment]
@pdurbin sorry about that - thanks for catching it!

**Pull request merged**